### ISSUE TITLE
On New install(v1.2.7), php artisan app:install --env=production fails 

### ISSUE THREAD

**Issue by citizenkepler** [snipe/snipe-it#783]
Twice I have been able to replicate the following error when running the `php artisan app:install --env=production` command excuted as described in http://snipeitapp.com/documentation/installing-on-linuxosx/.

```
  [InvalidArgumentException]
  There are no commands defined in the "app" namespace.
```

PHP/5.5.24 (DSO Handler)
cPanel LAMP Stack
Apache MPM ITK
Server version: Apache/2.2.29 (Unix)

**Comment by snipe** [comment]
Can you show me the output of the composer install you did right before that?

**Comment by citizenkepler** [comment]
Hello,  Heres the composer install beforehand.  The only warning I saw was with d11wtq/boris

```
[~]# php composer.phar install --no-dev
X-Powered-By: PHP/5.5.24
Content-type: text/html

Warning: Composer should be invoked via the CLI version of PHP, not the cgi-fcgi SAPI
Loading composer repositories with package information
Installing dependencies from lock file
  - Installing symfony/finder (2.5.x-dev e527ebf)
    Downloading: 100%

  - Installing symfony/var-dumper (2.8.x-dev 2bb458c)
    Downloading: 100%

  - Installing psr/log (dev-master bf2c13d)
    Downloading: 100%

  - Installing maximebf/debugbar (dev-master 0d1fcdd)
    Downloading: 100%

  - Installing symfony/translation (2.5.x-dev 165b534)
    Downloading: 100%

  - Installing symfony/security-core (2.5.x-dev 3a27d7b)
    Downloading: 100%

  - Installing symfony/routing (2.5.x-dev 46142c3)
    Downloading: 100%

  - Installing symfony/process (2.5.x-dev 00a1308)
    Downloading: 100%

  - Installing symfony/http-foundation (2.5.x-dev 08e7838)
    Downloading: 100%

  - Installing symfony/event-dispatcher (2.8.x-dev 44bde96)
    Downloading: 100%

  - Installing symfony/debug (2.5.x-dev ed30195)
    Downloading: 100%

  - Installing symfony/http-kernel (2.5.x-dev 7b1632c)
    Downloading: 100%

  - Installing symfony/dom-crawler (2.5.x-dev 3860edc)
    Downloading: 100%

  - Installing symfony/css-selector (2.5.x-dev d45b306)
    Downloading: 100%

  - Installing symfony/console (2.5.x-dev a43e750)
    Downloading: 100%

  - Installing symfony/browser-kit (2.5.x-dev b2b78b8)
    Downloading: 100%

  - Installing swiftmailer/swiftmailer (5.x-dev 728eba2)
    Downloading: 100%

  - Installing stack/builder (dev-master 14afc80)
    Downloading: 100%

  - Installing predis/predis (0.8.x-dev 4123fcd)
    Downloading: 100%

  - Installing phpseclib/phpseclib (dev-master c0511f2)
    Downloading: 100%

  - Installing patchwork/utf8 (dev-master 47056e3)
    Downloading: 100%

  - Installing nesbot/carbon (1.17.0)
    Downloading: 100%

  - Installing monolog/monolog (dev-master d84ba2c)
    Downloading: 100%

  - Installing nikic/php-parser (0.9.x-dev ef70767)
    Downloading: 100%

  - Installing jeremeamia/superclosure (1.0.x-dev 4d89ca7)
    Downloading: 100%

  - Installing ircmaxell/password-compat (1.0.x-dev 9b99377)
    Downloading: 100%

  - Installing filp/whoops (1.1.5)
    Downloading: 100%

  - Installing d11wtq/boris (v1.0.8)
    Downloading: Connecting...    Failed to download d11wtq/boris from dist: The "https://api.github.com/repos/d11wtq/boris/zipball/125dd4e5752639af7678a22ea597115646d89c6e" file could not be downloaded (HTTP/1.1 404 Not Found)
    Now trying to download from source
  - Installing d11wtq/boris (v1.0.8)
    Cloning 125dd4e5752639af7678a22ea597115646d89c6e

  - Installing symfony/filesystem (2.8.x-dev 41b7d35)
    Downloading: 100%

  - Installing classpreloader/classpreloader (1.0.2)
    Downloading: 100%

  - Installing laravel/framework (v4.2.17)
    Downloading: 100%

  - Installing barryvdh/laravel-debugbar (1.x-dev d510da0)
    Downloading: 100%

  - Installing cartalyst/sentry (v2.1.5)
    Downloading: 100%

  - Installing chumper/datatable (2.3)
    Downloading: 100%

  - Installing dinesh/barcode (4.0.x-dev 5655ece)
    Downloading: 100%

  - Installing doctrine/lexer (dev-master 83893c5)
    Downloading: 100%

  - Installing doctrine/annotations (dev-master b5202eb)
    Downloading: 100%

  - Installing doctrine/cache (dev-master 9d14133)
    Downloading: 100%

  - Installing doctrine/inflector (dev-master e5eaf8c)
    Downloading: 100%

  - Installing doctrine/collections (v1.3.0)
    Downloading: 100%

  - Installing doctrine/common (dev-master 5add480)
    Downloading: 100%

  - Installing doctrine/dbal (v2.4.2)
    Downloading: 100%

  - Installing erusev/parsedown (dev-master b9e5228)
    Downloading: 100%

  - Installing react/promise (v2.2.0)
    Downloading: 100%

  - Installing guzzlehttp/streams (dev-master d1f8a6c)
    Downloading: 100%

  - Installing guzzlehttp/ringphp (dev-master 2f75325)
    Downloading: 100%

  - Installing intervention/image (dev-master 6626d76)
    Downloading: 100%

  - Installing league/csv (dev-master f17b721)
    Downloading: 100%

  - Installing guzzlehttp/guzzle (5.2.0)
    Downloading: 100%

  - Installing maknz/slack (dev-master 9c4f7d0)
    Downloading: 100%

  - Installing symfony/config (2.8.x-dev b9a3893)
    Downloading: 100%

  - Installing symfony/stopwatch (2.8.x-dev 20c8903)
    Downloading: 100%

  - Installing symfony/yaml (2.8.x-dev 4ededb1)
    Downloading: 100%

Generating autoload files
NOTICE: Since laravel-debugbar 1.7.x, publishing assets is no longer necessary. The assets in public/packages/barryvdh/laravel-debugbar and maximebf/php-debugbar can be safely removed.
X-Powered-By: PHP/5.5.24
Content-type: text/html

 [~]# php artisan app:install --env=production



  [InvalidArgumentException]
  There are no commands defined in the "app" namespace.



X-Powered-By: PHP/5.5.24
Content-type: text/html

[~]# cd Write failed: Broken pipe
```

**Comment by snipe** [comment]
So, it looks like your composer installed okay, but this bit afterwards I s weird:

X-Powered-By: PHP/5.5.24
Content-type: text/html

I need to look into this further. I haven't seen this issue before.

If you install 1.2.6.1, do you run into the same problem?

**Comment by citizenkepler** [comment]
I have installed 1.2.6.1 sucessfully, the only differnce besides the version number is use /usr/local/bin/php as this is a cPanel system and the path default /usr/bin/php passes commandline execution through a php handler causing the `X-Powered-By: PHP/5.5.24` headers and this warning `Warning: Composer should be invoked via the CLI version of PHP, not the cgi-fcgi SAPI`

After the successfull install I am encountering a 500 error.

I have the install log here http://citizenkepler.com/offsitelink/install.txt

Thanks, when I get back home I will try a 1.2.7 install, but using the /usr/local/bin/php binary so that install's php does not pass through a php handler (like cPanel likes to do).

Thanks,
~Citizen

**Comment by snipe** [comment]
What do your all logs say?

**Comment by snipe** [comment]
*app logs

**Comment by snipe** [comment]
(My guess is that you didn't change your application key in `app/config/production/app.php`)
http://snipeitapp.com/documentation/common-issues/

**Comment by snipe** [comment]
Any update here?

**Comment by citizenkepler** [comment]
I've been in a busy spell the past 3 days and haven't been able to look into it.  I will update this issue tomorrow morning when I believe I have bit to finish this up. 

Thanks for your help.

**Comment by citizenkepler** [comment]
Hello,

I have been able to get v1.2.6-26 installed and functional. What I have learned from this is that on specifically cPanel the php in $PATH runs /usr/bin/php which has a php handler wrapper that caused the install failures and ```X-Powered-By: PHP/5.5.24```lines.

Using /usr/local/bin/php calls the correct command-line version of php on cPanel that allows composer/etc to function normally. 

Thanks for the help, the last thing was the application key indeed.
